main.js

```diff
@@ -1,352 +1 @@
-
-        //Get players
-        const player1 = gameFlow.players.player1;
-        const player2 = gameFlow.players.player2;
-
-        // Find tile index from button id
-        const index = e.target.getAttribute('id').substring(1);
-        // Ensure space is available
-        if (!gameBoard.isAvailable(index)){
-            return;
-        }
-        // Change background color
-        e.target.className = 'selectedTile';
-        // If 1-player game
-        if (player2.name === 'computer') {
-
-            // Add mark to game tile & update counter
-            gameBoard.showIcon(index, player1.mark);
-            counter.add()
-
-            // Begin checking for winner if move count >= 5
-            if (counter.count >= 5) {
-                if (gameFlow.hasWon()){
-                    gameFlow.returnWinner(player1.mark);
-                    gameFlow.resetGame();
-                    return;
-                }
-                if (counter.count === 9) {
-                    // Check for tie
-                    if (gameFlow.hasTie()) {
-                        gameFlow.resetGame();
-                        return;
-                    }
-                } 
-            }
-
-            // Get computer move
-            let computerIndex = gameFlow.computerMove();
-            gameBoard.showIcon(computerIndex, player2.mark);
-            gameBoard.tiles[computerIndex].className = 'selectedTile';
-            counter.add()
-            //Check for winner again
-            if (counter.count >= 5) {
-                if (gameFlow.hasWon()){
-                    gameFlow.returnWinner(player2.mark);
-                    gameFlow.resetGame();
-                    return;
-                }
-                if (counter.count === 9) {
-                    // Check for tie
-                    if (gameFlow.hasTie()) {
-                        gameFlow.resetGame();
-                        return;
-                    }
-                } 
-            }
-            return;
-        }
-
-        // If 2-Player game, check whose move it is
-        let currentPlayer;
-        if (counter.count === 0 || counter.count % 2 === 0 ){
-            currentPlayer = player1;
-        } else {
-            currentPlayer = player2;
-        }
-
-        // Add mark to game tile & update counter
-        gameBoard.showIcon(index, currentPlayer.mark);
-        counter.add()
-
-        // Begin checking for winner if move count >= 5
-        if (counter.count >= 5) {
-            if (gameFlow.hasWon()){
-                gameFlow.returnWinner(currentPlayer.mark);
-                gameFlow.resetGame();
-                return;
-            }
-            if (counter.count === 9) {
-                // Check for tie
-                if (gameFlow.hasTie()) {
-                    gameFlow.resetGame();
-                    return;
-                }
-            } 
-        }
-
-        // If player2 is computer
-        return;
-    },
-    
-    hasWon()  {
```

Comment:
this is fine for a typical implementation of the game
It is worth considering how you could make the game more dynamic.
 ie - what if I want to play on a 4x4 board, or 900x900?
 How can we programmatically determine the winner, without hard-coded conditions?